docs/README.md

```diff
@@ -168,6 +144,102 @@ That's all you need to do. The `versions` directory is listed on server start to
 Because the navbar is automatically generated from the directory structure, the default ordering of the links under each section is alphabetical. However, for many sections, this is not ideal UX.
 So, if you wish to override the alphabetical ordering, manipulate page titles in **constants/navigation.js**.
 
+### Updating API reference docs
+
+The API reference docs are generated from the TypeScript source code.
+
+This section walks through the process of updating documentation for an Expo package. Throughout this document, we will assume we want to update TypeDoc definitions of property inside `expo-constants` as an example.
+
+#### Prerequisites
+
+Before proceeding, make sure you:
+
+- have [**expo/**](https://github.com/expo/expo) repo cloned on your machine
+  - make sure to [install `direnv`](https://direnv.net/docs/installation.html) and run `direnv allow` at the root of the **expo/** repo.
+- have gone through the steps mentioned in [**"Download and Setup" in the contribution guideline**](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-download-and-setup).
+- can run **expo/docs** repo **[locally](https://github.com/expo/expo/tree/main/docs#running-locally)**.
+
+Once you have made sure the development setup is ready, proceed to the next section:
+
+#### Step 1: Update the package’s TypeDoc
+
+- After you have identified which package docs you want to update, open a terminal window and navigate to that package’s directory. For example:
+
+```shell
+# Navigate to expo-constants package directory inside expo/ repo
+cd expo/packages/expo-constants
+```
+
+- Then, open **.ts** file in your code editor/IDE where you want to make changes/updates.
+- Start the TypeScript build compilation in watch mode using `yarn build` in the terminal window.
+- Make the update. For example, we want to update the TypeDoc description of [`manifest2` property](https://docs.expo.dev/versions/latest/sdk/constants/#nativeconstants)
+
+  - Inside the **src/** directory, open **Constants.types.ts** file.
+  - Search for `manifest2` property. It has a current description as shown below:
+
+  ```ts
+  /**
+   * New manifest for Expo apps using modern Expo Updates from a remote source, such as apps that
+   * use EAS Update. Returns `null` in bare workflow and when `manifest` is non-null.
+   * > Prefer using `Constants.expoConfig` instead, which behaves more consistently across classic
+   * updates and modern Expo Updates.
+   */
+  manifest2: Manifest | null;
+  ```
```

Comment:
I believe we can select simpler and more long-term example, since `manifest2` is a temporary field, a migration aid.